xcm/xcm-executor/src/assets.rs

```diff
@@ -190,6 +251,92 @@ impl Assets {
 			match asset {
 				MultiAsset::None => (),
 				MultiAsset::All => return self.swapped(Assets::default()),
+				MultiAsset::AllFungible => {
+					// Copy all fungible assets into result.
+					for (id, amount) in self.fungible.clone().into_iter() {
```

Comment:
You here take a reference to a vector, then create a new vector, then copy all the elements into the new vector, and then destruct the vector using `.into_iter` to iterate on every element.

I think a more idiomatic and efficient way to do the same is to create an iterator from a reference to a vector using `.iter` and then apply the `.cloned` combinator. 

```suggestion
					for (id, amount) in self.fungible.iter().cloned() {
```